### PR TITLE
Create a User through a mutation

### PR DESCRIPTION
**Pull request by whyderrick**
# Create a User
This commit establishes the first API action in the app: creating a user. To do so, it adds several things.

## What's Added
- It installs [Clearance](https://github.com/thoughtbot/clearance) for user authentication.
- It instantiates a user model
- It installs the [ruby-graphql gem](https://github.com/rmosolgo/graphql-ruby) and runs its installer to provide a GraphQL API layer.

## What's the New Behavior
With these building blocks, this commit defines our first mutation, `registerUser`, and returns either the id of a user or errors if creation failed due to a server validation.

## What Else is Noteworthy
There are a couple of architectural choices worth calling out here too. The mutation itself includes User validation errors on the object in the data, consistent with the [Errors as Data](https://graphql-ruby.org/mutations/mutation_errors#errors-as-data) pattern for GraphQL API design. For this app, the top level errors key is being reserved for server errors

### CODE REVIEW

File: spec/graphql/register_user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+require "rails_helper"
+
+RSpec.describe "GraphQL user mutations", type: :graphql do
+  describe "user creation" do
+    context "when given a new user name and password" do
+      it "creates a new user" do
+        user = build(:user)
+        mutation = <<~MUTATION
```
Comment (seanpdoyle) [inline spec/graphql/register_user_spec.rb]: Did you know that GitHub supports syntax highlighting heredoc strings based on the language specified in the delimiters? I use a Vim plugin to support that as well.

```rb
<<~GRAPHQL
($email:String!, $password:String!) {
  registerUser(email:$email, password: $password) {
    user {
      email
    }
  }
}
GRAPHQL
```

Comment (whyderrick) [reply spec/graphql/register_user_spec.rb]: I didn't. What's the plugin?

Comment (seanpdoyle) [reply spec/graphql/register_user_spec.rb]: https://github.com/joker1007/vim-ruby-heredoc-syntax

**Comment by whyderrick** [comment]
@mike-burns That was the goal. I'm using Vanda as a reference implementation for the most part. If there are regrettable decisions that y'all haven't corrected yet, let me know please.

**Pull request merged**